compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp

```diff
@@ -359,10 +391,8 @@ static bool hasCompatibleOuterParallelLoops(
   llvm::SmallBitVector consumerParallelLoops =
       getOuterParallelLoops(cast<TilingInterface>(consumer.getOperation()));
 
-  if (allowConsumerParallelismPessimization) {
-    if (producerParallelLoops.count() > consumerParallelLoops.count())
```

Comment:
Do we want to fuse even if `producerParallelLoops.count() > consumerParallelLoops.count()`? This condition seems to be removed.

Reply:
The overall logic is changed as follows
1) Now when we find the "root operation" of the dispatch we record the outer parallel loop structure of the op. Then for each op fused into the dispatch, we check if the outer parallel loop structure of this op matches the root op.
2) The "match" is done by
   (a) if the pulled in op is fully parallel, as long as it has many parallel loops as the root op, it is pulled in
   (b) if the pulled in op is not fully parallel (has reductions), then the outer parallel loop structure has to match exactly the outer parallel loop structure of the root op.

Impact is earlier we would only fuse consumer ops that are all parallel but might have more parallel loops than the root. Now we do that with the producer too.